slu-scripts/src/sensor-data/sensor-data.ts

```diff
@@ -1,36 +1,93 @@
 import fs from 'fs';
 import { ClientConfig, ChannelClient, ChannelData } from 'iota-is-sdk';
 import { createKey, decrypt } from '../vpuf/vpuf';
-import { Axios } from 'axios';
-
-const axios = new Axios();
+import { AxiosResponse } from 'axios';
+import { decryptData, sendAuthProof } from '../auth-proof/auth-proof';
+import crypto from 'crypto';
+import * as ed from '@noble/ed25519';
+import * as bs58 from 'bs58';
+import axios from 'axios';
 
 export async function sendData(
 	encryptedDataPath: string | undefined,
 	keyFilePath: string | undefined,
 	isConfigPath: string | undefined,
 	collectorBaseUrl: string | undefined,
-	payloadData: any
+	payloadData: any,
+	isAuthUrl: string | undefined,
+	jwt: string | undefined
 ): Promise<ChannelData> {
-	if (isConfigPath && encryptedDataPath && keyFilePath && collectorBaseUrl) {
+	if (isConfigPath && encryptedDataPath && keyFilePath && collectorBaseUrl && isAuthUrl && jwt) {
 		const encryptedData = fs.readFileSync(encryptedDataPath, 'utf-8');
 		const key = createKey(keyFilePath);
 		const decryptedData = decrypt(encryptedData, key);
-		const { identity, channelAddress } = JSON.parse(decryptedData);
+		let { identityKeys, channelAddress } = JSON.parse(decryptedData);
```

Comment:
it's the format of the returned DTO of the one-shot-device and the identityKeys include id and key as an object. So to refactor the one-shot-device I would prefer to open another ticket because I don't know to which extend this refactoring will go because I didn't code the one-shot-device.